BootloaderCommonPkg/Library/BootloaderCommonLib/BootloaderCommonLib.c

```diff
@@ -359,22 +355,67 @@ GetComponentInfoByPartition (
       //
       if ( ((EntryDesc.Flags & (FLASH_MAP_FLAGS_NON_REDUNDANT_REGION | FLASH_MAP_FLAGS_NON_VOLATILE_REGION)) != 0) ||
            (((IsBackupPartition ? FLASH_MAP_FLAGS_BACKUP : 0) ^ (EntryDesc.Flags & FLASH_MAP_FLAGS_BACKUP)) == 0) ) {
-        PcdBase = (UINT32) (RomBase + EntryDesc.Offset);
-        PcdSize = EntryDesc.Size;
+        CopyMem((VOID *)Entry, (VOID *)&EntryDesc, sizeof(FLASH_MAP_ENTRY_DESC));

```

Comment:
Do we really need to copy ?   Why not return the entry in flash map directly ?

Reply:
No, I changed the code